core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestUnwrapRowDereference.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import io.trino.sql.planner.assertions.PlanMatchPattern;
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.plan.Assignments;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.project;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestUnwrapRowDereference
+        extends BaseRuleTest
+{
+    @Test
+    public void test()
+    {
+        test("ROW(1)[1]", "1");
+        test("ROW(1, 2)[1]", "1");
+        test("ROW(ROW(1, 2), 3)[1][2]", "2");
+
+        test("CAST(ROW(1, 2) AS row(a bigint, b bigint))[1]", "CAST(1 AS bigint)");
+        test("CAST(ROW(1, 2) AS row(bigint, bigint))[1]", "CAST(1 AS bigint)");
+    }
+
+    private void test(String original, String unwrapped)
+    {
+        tester().assertThat(new UnwrapRowDereference().projectExpressionRewrite())
+                .on(p -> p.project(
+                        Assignments.builder()
+                                .put(p.symbol("output", BOOLEAN), expression(original))
```

Comment:
Why `BOOLEAN`?

Reply:
That's left-over from something I had before. I'll fix it.